src/components/shared/form_fields/DirectDebitField.vue

```diff
@@ -0,0 +1,132 @@
+			:show-error="showAccountNumberError"
+			:help-text="$t( 'donation_form_payment_bankdata_account_iban_help_text' )"
+			@field-changed="$emit( 'field-changed', 'account-number' )"
+			@input="onAccountNumberInput"
+		>
+			<template #message>
+				<span v-if="bankName !== '' && accountNumberType === AccountNumberType.IBAN" class="iban-bank-name">{{ bankName }} ({{ bic }})</span>
+			</template>
+		</TextValueField>
+
+		<TextField
+			v-if="accountNumberType === AccountNumberType.Account"
+			v-model="bankCode"
+			input-id="bank-code"
+			name="bank-code"
+			:label="$t( 'donation_form_payment_bankdata_bank_legacy_label' )"
+			:placeholder="$t( 'donation_form_payment_bankdata_bank_legacy_placeholder' )"
+			error-message=""
+			:show-error="showBankCodeError"
+			:help-text="$t( 'donation_form_payment_bankdata_bank_legacy_help_text' )"
+			@field-changed="$emit( 'field-changed', 'bank-code' )"
+			@update:model-value="onBankCodeUpdate"
+		>
+			<template #message>
+				<span v-if="bankName !== ''" class="bank-name">{{ bankName }}</span>
+			</template>
+		</TextField>
+	</div>
+</template>
+
+<script setup lang="ts">
+
+import TextField from '@src/components/shared/form_fields/TextField.vue';
+import { computed, nextTick, ref, watch } from 'vue';
+import { BankValidationResource } from '@src/api/BankValidationResource';
+import TextValueField from '@src/components/shared/form_fields/TextValueField.vue';
+import { useFieldModel } from '@src/components/shared/form_fields/useFieldModel';
+import { AccountNumberType } from '@src/view_models/BankAccount';
+
+interface Props {
+	accountNumber: string;
+	bankCode: string;
+	bankName: string;
+	bic: string;
+	accountNumberType: AccountNumberType
+	showAccountNumberError: boolean;
+	showBankCodeError: boolean;
+	bankValidationResource: BankValidationResource;
+}
+
+const getDisplayValue = ( newValue: string ) => {
+	return newValue.replace( /(.{4})/g, '$& ' ).trim();
+};
+const props = defineProps<Props>();
+const emit = defineEmits( [ 'field-changed', 'update:accountNumber', 'update:bankCode' ] );
+
+const accountNumber = useFieldModel<string>( () => props.accountNumber, props.accountNumber );
+const bankCode = useFieldModel<string>( () => props.bankCode, props.bankCode );
+const formattedBankNumber = ref<string>( getDisplayValue( accountNumber.value ) );
+const bankNumberCursorPosition = ref<number>( 0 );
+
+const accountNumberLabel = computed<string>( () => {
+	if ( props.accountNumberType === AccountNumberType.IBAN ) {
+		return 'donation_form_payment_bankdata_account_iban_label';
+	} else if ( props.accountNumberType === AccountNumberType.Account ) {
+		return 'donation_form_payment_bankdata_account_legacy_label';
+	}
+	return 'donation_form_payment_bankdata_account_default_label';
+} );
+
+/**
+ * 1. Get the text before the cursor
+ * 2. Clear the spaces
+ * 3. Add new spaces in the correct positions, but don't trim
+ * 4. Return the length
+ *
```

Comment:
When I tested editing, it worked when inserting numbers and hitting backspace, but I got the jumping behavior when hitting the <kbd>Del</kbd> (forward delete) key